includes/Admin/Admin_AJAX.php

```diff
@@ -278,6 +223,70 @@ public function run_checks() {
 		);
 	}
 
+
+	/**
+	 * Get the AJAX_Runner runner.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return AJAX_Runner
+	 */
+	protected function get_runner() {
+		$runner = Plugin_Request_Utility::get_runner();
+
+		if ( is_null( $runner ) ) {
+			Plugin_Request_Utility::initialize_runner();
+			$runner = Plugin_Request_Utility::get_runner();
+		}
+
+		// Make sure we are using the correct runner instance.
+		if ( ! ( $runner instanceof AJAX_Runner ) ) {
+			wp_send_json_error(
+				new WP_Error( 'invalid-runner', __( 'AJAX Runner was not initialized correctly.', 'plugin-check' ) ),
+				500
+			);
```

Comment:
Similar to my other comment, we may want to return the `WP_Error` rather than send the JSON response here. (see https://github.com/10up/plugin-check/pull/97#discussion_r1136040211)

Alternatively, we could throw an `Exception` instead and call `get_runner` in the try/catch block?

Reply:
This is just the same logic that was there before. Why change it?

Reply:
See https://github.com/10up/plugin-check/pull/157/files#r1180537801, the utility methods should have the responsibility of what their name is, not return a JSON response. For example this method should return the runner, or throw an exception if there is an error. The AJAX callbacks can then use `try { ... } catch` and return a JSON error response in case of an exception.

Reply:
If we do that, that could mean repeated the same code every time this method is called. I am trying avoid lots of boilerplate code here.